python-connectors/neo4j/connector.py

```diff
@@ -66,9 +65,18 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None, partitio
                     yield {
                         "source node": rel.start_node["name"],
                         "relationship": type(rel).__name__,
-                        "target node": rel.end_node["name"]
+                        "target node": rel.end_node["name"],
                     }
 
+        elif self.config["select_nodes_or_relationships"] == "custom_query":
+            custom_query = self.config.get("selected_custom_query", None)
+            logging.info(f"Neo4j plugin - Dataset - User custom query: {custom_query}")
+            if records_limit > -1:
+                custom_query = f"CALL {{{custom_query}}} RETURN * LIMIT {records_limit}"
+            results = self.graph.run(custom_query)
+            for record in results.data():
```

Comment:
Just to make sure: there is no pagination mechanism ?

Reply:
I will look for more efficient way to retrieve large data

Reply:
Using `for row in session.run(query)` to get the rows one by one from the neo4j server.